File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/CS/MessagingTestCaseBase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

#if !SILVERLIGHT
using Db4oUnit;
using Db4oUnit.Extensions.Fixtures;
using Db4objects.Db4o;
using Db4objects.Db4o.CS;
using Db4objects.Db4o.CS.Config;
using Db4objects.Db4o.CS.Internal.Config;
using Db4objects.Db4o.Config;
using Db4objects.Db4o.Foundation;
using Db4objects.Db4o.IO;
using Db4objects.Db4o.Messaging;

namespace Db4objects.Db4o.Tests.Common.CS
{
	public class MessagingTestCaseBase : ITestCase, IOptOutMultiSession
	{
		public sealed class MessageCollector : IMessageRecipient
		{
			public readonly Collection4 messages = new Collection4();

			public void ProcessMessage(IMessageContext context, object message)
			{
				messages.Add(message);
			}
		}

		protected virtual IMessageSender MessageSender(IObjectContainer client)
		{
			return client.Ext().Configure().ClientServer().GetMessageSender();
		}

		protected virtual IObjectContainer OpenClient(string clientId, IObjectServer server
			)
		{
			server.GrantAccess(clientId, "p");
			return Db4oClientServer.OpenClient(MultithreadedClientConfig(), "127.0.0.1", server
				.Ext().Port(), clientId, "p");
		}

		private IClientConfiguration MultithreadedClientConfig()
		{
			IClientConfiguration config = Db4oClientServer.NewClientConfiguration();
			config.Networking.SingleThreadedClient = false;
			return config;
		}

		protected virtual IObjectServer OpenServerWith(IMessageRecipient recipient)
		{
			IConfiguration config = MemoryIoConfiguration();
			SetMessageRecipient(config, recipient);
			return OpenServer(config);
		}

		protected virtual IConfiguration MemoryIoConfiguration()
		{
			IConfiguration config = Db4oFactory.NewConfiguration();
			config.Storage = new MemoryStorage();
			return config;
		}

		protected virtual IObjectServer OpenServer(IConfiguration config)
		{
			return Db4oClientServer.OpenServer(Db4oClientServerLegacyConfigurationBridge.AsServerConfiguration
				(config), "nofile", unchecked((int)(0xdb40)));
		}

		protected virtual void SetMessageRecipient(IObjectContainer container, IMessageRecipient
			 recipient)
		{
			SetMessageRecipient(container.Ext().Configure(), recipient);
		}

		private void SetMessageRecipient(IConfiguration config, IMessageRecipient recipient
			)
		{
			config.ClientServer().SetMessageRecipient(recipient);
		}
	}
}
#endif // !SILVERLIGHT

File: db4oj.tests/src/com/db4o/db4ounit/common/cs/MessagingTestCaseBase.java
/* Copyright (C) 2004 - 2006 Versant Inc. http://www.db4o.com */

package com.db4o.db4ounit.common.cs;

import com.db4o.*;
import com.db4o.config.*;
import com.db4o.cs.config.*;
import com.db4o.cs.internal.config.*;
import com.db4o.foundation.*;
import com.db4o.io.*;
import com.db4o.messaging.*;

import db4ounit.*;
import db4ounit.extensions.fixtures.*;

public class MessagingTestCaseBase implements TestCase, OptOutMultiSession {
	
	public static final class MessageCollector implements MessageRecipient {
		public final Collection4 messages = new Collection4();
		
		public void processMessage(MessageContext context, Object message) {
			messages.add(message);
		}
	}

	protected MessageSender messageSender(final ObjectContainer client) {
		return client.ext().configure().clientServer().getMessageSender();
	}

	protected ObjectContainer openClient(String clientId, final ObjectServer server) {
		server.grantAccess(clientId, "p");
		
		return com.db4o.cs.Db4oClientServer.openClient(multithreadedClientConfig(), "127.0.0.1", server.ext().port(), clientId, "p");
	}

	private ClientConfiguration multithreadedClientConfig() {
		final ClientConfiguration config = com.db4o.cs.Db4oClientServer.newClientConfiguration();
		config.networking().singleThreadedClient(false);
		return config;
	}

	protected ObjectServer openServerWith(final MessageRecipient recipient) {
		final Configuration config = memoryIoConfiguration();
		setMessageRecipient(config, recipient);
		return openServer(config);
	}

	protected Configuration memoryIoConfiguration() {
		final Configuration config = Db4o.newConfiguration();
		config.storage(new MemoryStorage());
		return config;
	}

	protected ObjectServer openServer(final Configuration config) {
		return com.db4o.cs.Db4oClientServer.openServer(Db4oClientServerLegacyConfigurationBridge.asServerConfiguration(config), "nofile", 0xdb40);
	}

	protected void setMessageRecipient(final ObjectContainer container, final MessageRecipient recipient) {
		setMessageRecipient(container.ext().configure(), recipient);
	}

	private void setMessageRecipient(final Configuration config, final MessageRecipient recipient) {
		config.clientServer().setMessageRecipient(recipient);
	}

}